tests/net/socket/tcp/src/main.c

```diff
@@ -723,6 +723,108 @@ void test_open_close_immediately(void)
 	k_sleep(TCP_TEARDOWN_TIMEOUT);
 }
 
+#define TCP_CLOSE_FAILURE_TIMEOUT 60000
+
+void test_close_obstructed(void)
+{
+	/* Test if socket closing even when there is not communication
+	 * possible any more
+	 */
+	uint32_t start_time, time_diff;
+	int count_before = 0, count_after = 0;
+	struct sockaddr_in c_saddr;
+	struct sockaddr_in s_saddr;
+	struct sockaddr addr;
+	socklen_t addrlen = sizeof(addr);
+	int c_sock;
+	int s_sock;
+	int new_sock;
+	int dropped_packets_before = 0;
+	int dropped_packets_after = 0;
+
+	prepare_sock_tcp_v4(CONFIG_NET_CONFIG_MY_IPV4_ADDR, ANY_PORT,
+			    &c_sock, &c_saddr);
+	prepare_sock_tcp_v4(CONFIG_NET_CONFIG_MY_IPV4_ADDR, SERVER_PORT,
+			    &s_sock, &s_saddr);
+
+	test_bind(s_sock, (struct sockaddr *)&s_saddr, sizeof(s_saddr));
+	test_listen(s_sock);
+
+	zassert_equal(connect(c_sock, (struct sockaddr *)&s_saddr,
+				sizeof(s_saddr)),
+				0, "connect not succeed");
+	test_accept(s_sock, &new_sock, &addr, &addrlen);
+
+	/* We should have two contexts open now */
+	net_context_foreach(calc_net_context, &count_before);
+
+	/* Break the communication */
+	loopback_set_packet_drop_ratio(1.0f);
+
+	dropped_packets_before = loopback_get_num_dropped_packets();
+
+	test_close(c_sock);
+
+	k_sleep(K_MSEC(10));
+
+	dropped_packets_after = loopback_get_num_dropped_packets();
+	int dropped_packets = dropped_packets_after - dropped_packets_before;
+
+	/* At least some packet should have been dropped */
+	zassert_equal(dropped_packets,
+			CONFIG_NET_TCP_RETRY_COUNT,
+			"Insuffcient FIN retries, got %i", dropped_packets);
+
+	start_time = k_uptime_get_32();
+
+	/* After the client socket closing, the context count should be 1 */
```

Comment:
Actually, this is also not true, we still have two sockets allocated at this point (the listening `s_socks_sock and `new_sock` pending disconnection).

Reply:
You are right, change to: After the client socket closing, the context count should be 1 less